### ISSUE TITLE
Should we provide a set of code recipes?

### ISSUE THREAD

**Issue by dcodeIO** [AssemblyScript/meta#11]
So, lately, [this happened](https://twitter.com/kchplr/status/1152655652449390597), and the answer got me thinking about those recipes - like very short examples showcasing one particular aspect without the bloat. This is very similar to @torch2424's Wasm By Example and @MaxGraey's "Did you know?" AssemblyScript snippets, so I thought I pin this here for discussion. Not quite sure how to present this, but having an additional category right in the docs seems good so the recipes are easy to discover for new users. Thoughts? :)

**Comment by torch2424** [comment]
@dcodeIO So I'm most likely biased, but maybe it'll be a good idea to link to WasmByExample in the docs? :smile:

**Comment by dcodeIO** [comment]
Yeah, should definitely do that! :) Now I'm wondering if there are other starter questions that are so very AS-specific that Wasm By Example would not cover them. Might well be that there are not. Hard to tell without knowing the questions.

**Comment by dcodeIO** [comment]
@torch2424 Is there an URL I can use to preselect AssemblyScript on Wasm By Example?

**Comment by torch2424** [comment]
@dcodeIO 

> Yeah, should definitely do that! :) Now I'm wondering if there are other starter questions that are so very AS-specific that Wasm By Example would not cover them. Might well be that there are not. Hard to tell without knowing the questions.

So I guess it depends on what the person is looking for? If they are doing more internal specifc stuff, like building specific features or mainting a code base, then they probably want the AS docs.

But if they are someone super new, that wants to give wasm a try, or just needs a quick example or starting point, wasmbyexample makes sense (in my opinion).

But yeah it's definitely hard to tell. I would maybe just put a small blurb about it in the examples intro, or an additional resources page? :thinking: 

> Is there an URL I can use to preselect AssemblyScript on Wasm By Example?

Oh oops. Yeah I should totally allow that. I opened: https://github.com/torch2424/wasm-by-example/issues/40 :smile:

**Comment by dcodeIO** [comment]
Those are all good points, yeah. So perhaps, with Wasm By Example being prominently placed on the Examples page now, we should observe whether and which additional needs come up in the near future, and continue the discussion from there, as one possible outcome is that the questions are so very specific that adding example inline on the respective docs pages is the way to go.

**Comment by torch2424** [comment]
> Those are all good points, yeah. So perhaps, with Wasm By Example being prominently placed on the Examples page now, we should observe whether and which additional needs come up in the near future, and continue the discussion from there

Totally agree 😄 We can always continue the discussion and add/move/remove as needed 😄 

> as one possible outcome is that the questions are so very specific that adding example inline on the respective docs pages is the way to go.

Yeah definitely. Like for questions like: "What are all the methods on the String Object in AssemblyScript, and how do I split / concatenate them?" Is something I don't see WasmByExample a great fit for (other than a general introduction to Strings in AssemblyScript), whereas it would be a perfect fit for the docs 😄 

Also, I'll try and do some wasmbyexample cleanup soon 😄 And make it so you can point directly to AssemblyScript 👍

**Comment by torch2424** [comment]
@dcodeIO I fixed: https://github.com/torch2424/wasm-by-example/issues/40

Now you should be able to navigate directly to assemblyscript examples using: [https://wasmbyexample.dev/?programmingLanguage=assemblyscript](https://wasmbyexample.dev/?programmingLanguage=assemblyscript). You can try it out, by visiting that link, setting your language to rust. Then, clicking that link again and you should pop into AssemblyScript :smile:
